### PR TITLE
Add `--icp=CYCLE_POINT` option for various commands

### LINKED ISSUE #1000: initial cycle point on command line only?
https://github.com/cylc/cylc/pull/997#issuecomment-47302861
https://github.com/cylc/cylc/pull/997#issuecomment-47328047

Pre cylc-6, initial cycle time (ICT) was optional in the suite definition; it could be given or overridden on the command line for suite start-up.  Other suite parsing commands (e.g. validate) did not need the CLI option because a sensible default ICT could always be chosen.  In other words, under the old cycling paradigm a suite's validity could be determined without reference to a concrete initial cycle time.

For cylc-6 this isn't the case - a sensible default ICPoint can't be determined in general. So our options are:
1. retain the CLI option and give it to all suite parsing commands, or 
2. **require** an ICP in the suite definition

My preference is to require an ICP in the suite definition.  It is simpler, and I don't like the idea of suite definitions not being valid until extra external information is supplied.  I expect we have some users who rely on the command line ICT, but: cylc-6 changes the game a bit; it's not hard to edit a suite definition; and if really necessary a Jinja2 variable can be set from the command line (and that's already supported by all suite-parsing commands, I think).

Thoughts?

### PR DESCRIPTION
**Pull request by matthewrmshin**
Allow initial cycle point to be specified if missing from `suite.rc`.

Close #1000.

### CODE REVIEW

**Comment by hjoliver** [comment]
@matthewrmshin - is this ready to go as far as you're concerned? (I might have done a premature review, can't recall if you assigned me or not).  Assuming it is done, I'll assign @oliver-sanders for a second review...

**Comment by matthewrmshin** [comment]
Should be good to go. Had fixed a conflict last night by re-basing.

File: tests/cylc-diff/01-icp.t [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# Test for "cylc diff --icp=CYCLE_POINT".
+. "$(dirname "$0")/test_header"
+
+set_test_number 3
+
+init_suite "${TEST_NAME_BASE}-1" <<'__SUITE_RC__'
+[cylc]
+    UTC mode = True
+[scheduling]
+    [[dependencies]]
```
Comment (oliver-sanders) [inline tests/cylc-diff/01-icp.t]: We could provide one of these suite defs with an ICP for a more comprehensive test.

Comment (matthewrmshin) [inline tests/cylc-diff/01-icp.t]: New override test added.

**Comment by matthewrmshin** [comment]
Branch updated, squashed and re-based.

**Pull request merged**